test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -415,7 +415,7 @@ func deleteVolumeSnapshot(ctx context.Context, f *framework.Framework, dc dynami
 	case storageframework.RetainSnapshot:
 		ginkgo.By("checking the SnapshotContent has not been deleted")
 		err = storageutils.WaitForGVRDeletion(ctx, dc, storageutils.SnapshotContentGVR, vscontent.GetName(), 1*time.Second /* poll */, 30*time.Second /* timeout */)
-		framework.ExpectError(err)
+		gomega.Expect(err).To(gomega.HaveOccurred())
```

Comment:
Using `gomega.HaveOccurred` doesn't improve the tests, it's just a different way of saying the same thing. What we want is `gomega.MatchError` with a more specific match against the error that should occur.

But in this case, using `gomega.Consistently` instead of "wait for something and expect it to fail" is much better. This implies improving `storageutils`.

Reply:
please review the new function “EnsureGVRDeletion” in storageutils. thanks.

Reply:
`EnsureGVRDeletion` uses `Eventually`. As mentioned above, what needs to be used is `Consistently`.

Reply:
@pohly `WaitForGVRDeletion` waits for an existing GVR to be deleted. `EnsureGVRDeletion` is implemented with `gomega` and used to replace `WaitForGVRDeletion`, It's not used to verify that a GVR never exists within a period of time. so using `Consistently` is reasonable.

Reply:
The old code is:
```
	err = storageutils.WaitForGVRDeletion(ctx, dc, storageutils.SnapshotContentGVR, vscontent.GetName(), 1*time.Second /* poll */, 30*time.Second /* timeout */)
	framework.ExpectError(err)
```

This passes if (and only if) the GVR is never seen as being deleted for 30 seconds. If it ever had been seen as deleted, `WaitForGVRDeletion` would have returned nil and `framework.ExpectError` would have failed.

Therefore we need `gomega.Consistently` to match the test semantic.

I know the existing code is confusing. It's basically a misuse of `WaitForGVRDeletion`. Whatever new function we add is not a direct replacement for `WaitForGVRDeletion`, it has to have a different semantic.

Reply:
I understand your idea. If so, we shouldn't use `framework.ExpectError(err)` or `gomega.Expect(err).To(gomega.HaveOccurred()` to check helper function error.

The changes should like this:

```go
	switch pattern.SnapshotDeletionPolicy {
	case storageframework.DeleteSnapshot:
		ginkgo.By("checking the SnapshotContent has been deleted")
		err = storageutils.HelperFunc1(ctx, dc, storageutils.SnapshotContentGVR, vscontent.GetName(), framework.Poll, f.Timeouts.SnapshotDelete)
		framework.ExpectNoError(err)
	case storageframework.RetainSnapshot:
		ginkgo.By("checking the SnapshotContent has not been deleted")
		err = storageutils.HelperFunc2(ctx, dc, storageutils.SnapshotContentGVR, vscontent.GetName(), 1*time.Second /* poll */, 30*time.Second /* timeout */)
		framework.ExpectNoError(err)
	}
```

HelperFunc1 uses `Eventually` and HelperFunc2 uses `Consistently`. we can not use the same function for two cases. what do you think? @pohly

Reply:
Correct.

One helper is `Wait*` (= return with success as soon as state reached), the other is `Ensure*` (= wait for a fixed time and fail if the state is ever not as expected).

Reply:
got it. thanks @carlory and @pohly

Reply:
add new function to handle "HelperFunc2"